### PR TITLE
releng - automated releases

### PR DESCRIPTION
**Pull request by kapilt**
full automated releases with basic cli exercises.

addresses #7196 

draft as it would be great to get feedback on this

todo:
 - [ ] wrap functional testing as reusable workflow here.
 - [ ] also needs a guard on extant release blocker issues.

builds on and includes #8392

### CODE REVIEW

File: tools/dev/poetrypkg.py [hunk 1]
```diff
@@ -12,6 +12,26 @@ def cli():
...
 
+def project_roots(root):
+    for config_path in Path(root).rglob("pyproject.toml"):
+        yield config_path.parent
+
+
+@cli.command()
+@click.option('-r', '--root', type=click.Path())
+@click.option('-o', '--output', type=click.Path())
+def gen_test_requires(root, output):
```
Comment (kapilt) [inline tools/dev/poetrypkg.py:23]: s/test/release to make it a little more obvious

**Comment by kapilt** [comment]
moving from draft solely to try and trigger workflow for testing from pr.

**Comment by kapilt** [comment]
I think we're going to need to fine tune this once in the repo, for some reason gh workflow run --ref branch isn't working on this workflow (returns 404)

File: tools/dev/poetrypkg.py (hunk 1 above)
Comment (kapilt) [reply tools/dev/poetrypkg.py]: done and merged in the pr this originated from.

**Review by thisisshi — APPROVED** [review]

**Pull request merged**